starlite/events/emitter.py

```diff
@@ -49,10 +46,77 @@ async def emit(self, event_id: str, *args: Any, **kwargs: Any) -> None:  # pragm
         """
         raise NotImplementedError("not implemented")
 
+    @abstractmethod
+    async def on_startup(self) -> None:  # pragma: no cover
+        """Hook called on application startup, used to establish connection or perform other async operations.
+
+        Returns:
+            None
+        """
+        raise NotImplementedError("not implemented")
+
+    @abstractmethod
+    async def on_shutdown(self) -> None:  # pragma: no cover
+        """Hook called on application shutdown, used to perform cleanup.
+
+        Returns:
+            None
+        """
+        raise NotImplementedError("not implemented")
+
 
 class SimpleEventEmitter(BaseEventEmitterBackend):
     """Event emitter the works only in the current process"""
 
+    __slots__ = ("_queue", "_worker_task")
+
+    _queue: Queue | None
+    _worker_task: Task | None
+
+    def __init__(self, listeners: Sequence[EventListener]):
+        """Create an event emitter instance.
+
+        Args:
+            listeners: A list of listeners.
+        """
+        super().__init__(listeners=listeners)
+        self._queue = None
+        self._worker_task = None
+
+    async def _worker(self) -> None:
+        """Worker that runs in a separate thread and continuously pulls events from asyncio queue.
```

Comment:
```suggestion
        """Worker that runs in a separate task and continuously pulls events from asyncio queue.
```

it doesn't actually run in a separate thread